### PR TITLE
Add value flag to disable check-db-ready initContainers in Helm chart

### PR DESCRIPTION
**Pull request by easontm**
## Summary & Motivation

Address #26311 (partially) and #26284 (duplicate issues)

tl;dr: the `check-db-ready` container can be part of a race condition with other initContainers that are responsible for enabling the DB connection. Externally managed DBs (i.e. not the postgres container optionally included in this chart) can be validated before Helm installation so this step is not strictly required.

## How I Tested These Changes

1. Wrote some tests
2. Installed the chart to minikube with
```
helm install dagster . --set dagsterWebserver.image.tag=latest --set dagsterDaemon.image.tag=latest --set pipelineRun.image.tag=latest --set dagsterWebserver.checkDbReadyInitContainer=true
```
And observed the initContainers present. Then ran it again with `false`
```
helm install dagster . --set dagsterWebserver.image.tag=latest --set dagsterDaemon.image.tag=latest --set pipelineRun.image.tag=latest --set dagsterWebserver.checkDbReadyInitContainer=false
```
And observed the initContainers absent.

## Changelog

- Added flag `checkDbReadyInitContainer` to optionally disable db check initContainer.

### CODE REVIEW

File: helm/dagster/schema/schema_tests/test_celery_queues.py [hunk 1]
```diff
@@ -307,3 +307,35 @@ def test_scheduler_name(deployment_template: HelmTemplate):
...
+            type=RunLauncherType.CELERY,
+            config=RunLauncherConfig(
+                celeryK8sRunLauncher=CeleryK8sRunLauncherConfig.construct(
+                    checkDbReadyInitContainer=False
+                )
+            ),
+        )
+    )
+    [daemon_deployment] = deployment_template.render(helm_values)
+    assert daemon_deployment.spec.template.spec.init_containers is None or "check-db-ready" not in [
```
Comment (easontm) [inline helm/dagster/schema/schema_tests/test_celery_queues.py:324]: The first `is None` check is not strictly necessary for all tests, but default templates have initContainers and some don't. The shortcircuit makes it so the tests which _do_ have an initContainer by default will not start failing if that container is removed.

**Comment by cmpadden** [comment]
Hey @gibsondan, these changes look reasonable and well enough tested to me.

Do you know who would be best to give the final sign off, saw you in the git blame a bit, but there were a lot of people in there.

**Review by gibsondan — CHANGES_REQUESTED** [review]
Hi @easontm - thank you for adding this and for the thorough tests. I have one last request for you related to said thorough tests - can you add a case where the value is not explicitly set? I think it's important that we capture that nothing has changed if the user has not explciitly overridden the value and is relying on it being set to true in the default templated values.yaml (so that we know the check won't be inadvertently disabled).

For example, if you had accidentally left out something in the values.yaml (or somebody accidentally removed it in the future) it would be good if a test started failing.

Other than that hopefully relatively quick change this looks great to me.

File: helm/dagster/schema/schema_tests/test_celery_queues.py [hunk 2]
```diff
@@ -307,3 +307,35 @@ def test_scheduler_name(deployment_template: HelmTemplate):
...
+    ]
+
+    helm_values = DagsterHelmValues.construct(
+        runLauncher=RunLauncher(
+            type=RunLauncherType.CELERY,
+            config=RunLauncherConfig(
+                celeryK8sRunLauncher=CeleryK8sRunLauncherConfig.construct(
+                    checkDbReadyInitContainer=True
+                )
+            ),
```
[the remark this answers is not in the document]
Comment (easontm) [reply helm/dagster/schema/schema_tests/test_celery_queues.py:337]: Added! I verified by replacing `true` with `~` in the `values.yaml` file to intentionally break the test and it failed as it should.

**Review by gibsondan — APPROVED** [review]
thanks!
